pkg/test/local/proxy/envoy/discovery/interceptor.go

```diff
@@ -0,0 +1,36 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package discovery
+
+import xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+
+var (
+	// Identity is a global instance of the identity interceptor.
+	Identity = &IdentityResponseInterceptor{}
+)
+
+// ResponseInterceptor is an interceptor of discovery responses from the server, which allows the application
+// to modify the response returned to the client.
+type ResponseInterceptor interface {
+	InterceptDiscoveryResponse(resp *xdsapi.DiscoveryResponse) (*xdsapi.DiscoveryResponse, error)
```

Comment:
Nit: it looks like this simply be a lambda.

Reply:
Good point. Done.